src/imgui/ImGuiDebugger.cc

```diff
@@ -629,13 +629,39 @@ void ImGuiDebugger::drawDisassembly(CPURegs& regs, MSXCPUInterface& cpuInterface
 								}
 							});
 
+							enum Priority {
+								NO_MATCH = 0, // from lowest to highest
+								MISSING_ONE,
+								SLOT_AND_SEGMENT
+							};
+							Priority currentPriority = Priority::NO_MATCH;
+							std::vector<std::string_view> candidates;
+							auto add = [&](const Symbol* sym, Priority newPriority) {
+								if (newPriority < currentPriority) return; // we already have a better candidate
+								if (newPriority > currentPriority) candidates.clear(); // drop previous candidates, we found a better one
+								currentPriority = newPriority;
+								candidates.push_back(sym->name); // cycle symbols in the same priority level
+							};
+
+							auto slot = getCurrentSlot(cpuInterface, debugger, addr16);
+							auto psss = (slot.ss.value_or(0) << 2) + slot.ps;
 							auto addrLabels = symbolManager.lookupValue(addr16);
-							std::string_view displayAddr = addrLabels.empty()
-								? std::string_view(addrStr)
-								: std::string_view(addrLabels[cycleLabelsCounter % addrLabels.size()]->name);
+							for (const Symbol* symbol: addrLabels) {
+								if (symbol->slot) {
+									if (*symbol->slot != psss) continue; // slot mismatch -> don't show
+									if (symbol->segment) {
+										if (*symbol->segment != slot.seg) continue; // segment mismatch -> don't show
+										add(symbol, Priority::SLOT_AND_SEGMENT); // both match
+									} else {
+										add(symbol, Priority::MISSING_ONE); // only slot matches
+									}
+								} else if (symbol->segment == slot.seg) {
+									add(symbol, Priority::MISSING_ONE); // only segment matches
+								}
```

Comment:
I don't see any `add()` with `Priority::NO_MATCH`, that can't be right. Can I also ask how you tested this PR? Maybe in your next tests also check if symbols without slot/segment info still show up.

If we come in the else-branch of `if (symbol->slot) {` we don't yet know anything about the segment (not whether this info is present and not whether it matches). So maybe this should become something like this?
```
if (symbol->slot) {
    ...
} else if (symbol->segment) { // no slot info, but there is segment info
    if (*symbol->segment != slot.seg) continue; // segment mismatch -> don't show
    add( ... MISSING_ONE);
} else {
   add(... BOTH_MISSING);
}
```

BTW: maybe `MISSING_BOTH` is a better name than `NO_MATCH`?   (because if there is a mismatch (~no match) we won't include the symbol at all).

BTW2: In my previous comments/reviews I was wrong: I assumed segment isn't important when slot is missing (I even explicitly stated that as one of my assumptions). If you see such mistakes then please speak up. You don't have to apply or agree with all my requests for changes.

Reply:
Update:
The if-then-else logic above is again becoming quite complex. Maybe this alternative is simpler?
```
// skip symbols with any mismatch
if (symbol->slot && *symbol->slot != psss) continue;
if (symbol->segment && *symbol->segment != slot.seg) continue;

// the info that's present does match
if (symbol->slot && symbol->segment) {
    add(... SLOT_AND_SEGMENT);
} else if (!symbol->slot && !symbol->segment) {
    add(... MISSING_BOTH);
} else {
    add(... MISSING_ONE);
}
```
This is maybe also closer to what you started with in one of your previous revisions of this PR? So again sorry if I mislead you with my (wrong) comments.

Reply:
> I don't see any `add()` with `Priority::NO_MATCH`, that can't be right. Can I also ask how you tested this PR? Maybe in your next tests also check if symbols without slot/segment info still show up.

I tested it thoroughly this time and I think it's OK now.